### PR TITLE
Introduce onboarding wizard

### PR DESCRIPTION
**Pull request by shervElmi**
This PR introduces a new "Onboarding Wizard" feature within the Business Directory Plugin Admin Panel.

#### First Method:

Activate the Business Directory plugin from the WP Admin Plugins page. Please note that you will be redirected to the Onboarding Wizard page **only if this is your first time activating the plugin**. Otherwise, you can access it directly via the URL.

#### Second Method:

Directly access the Onboarding Wizard via the URL.
Example: `http://example.com/wp-admin/admin.php?page=wpbdp-onboarding-wizard`

### Demo Video
CleanShot 2024-11-04 at 19 25 27

### CODE REVIEW

File: includes/admin/controllers/class-onboarding-wizard.php [hunk 1]
```diff
@@ -0,0 +1,432 @@
...
+	/**
+	 * Validates if the Onboarding Wizard page is being displayed.
+	 *
+	 * @since x.x
+	 * @deprecated x.x
+	 *
+	 * @return bool True if the Onboarding Wizard page is displayed, false otherwise.
+	 */
+	public static function is_onboarding_wizard_displayed() {
+		_deprecated_function( __METHOD__, 'x.x' );
```
Comment (sorinmarta) [inline includes/admin/controllers/class-onboarding-wizard.php:429]: Why are we deprecating a method that we just added?

LE: I don't see any usage of this method in the current PR, so we might be safe just removing the function directly.

**Review by sorinmarta — APPROVED** [review]
Awesome work @shervElmi, thanks! I just left a few comments with suggestions, but I don't see anything that would break functionality.

File: assets/js/onboarding-wizard/events/collapsibleListener.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+/**
+ * Manages event handling for the "Skip" step button.
+ *
+ * @return {void}
+ */
+function addCollapsibleEvents() {
+	const { collapsible } = getElements();
+
+	collapsible.addEventListener( 'click', onCollapsibleClick );
+}
```
[the remark this answers is not in the document]
Comment (shervElmi) [reply assets/js/onboarding-wizard/events/collapsibleListener.js:16]: Hey @sorinmarta,

Thanks for the review and the kind words!

I’m using a modular pattern here, where this module functions as a listener, and `addCollapsibleEvents` acts as a constructor method. Following SOLID principles, specifically the Open/Closed principle, I aimed to design this function to be closed for modification but open for extension.

This setup allows us to easily expand the module in the future by adding listeners for other events, like `hover` or `change`, without altering the core logic. It maintains separation of concerns and keeps the module extensible for future requirements.

That said, if you strongly feel that using arrow functions for consistency is preferable, I'm open to adjusting it! 😊

[the remark this answers is not in the document]
Comment (sorinmarta) [reply assets/js/onboarding-wizard/events/collapsibleListener.js:16]: No worries, @shervElmi! Thanks for the explanation!

**Pull request merged**